Library/Homebrew/brew.sh

```diff
@@ -195,19 +195,30 @@ then
   HOMEBREW_BASH_COMMAND="$HOMEBREW_LIBRARY/Homebrew/dev-cmd/$HOMEBREW_COMMAND.sh"
 fi
 
-if [[ "$(id -u)" = "0" && "$(/usr/bin/stat -f%u "$HOMEBREW_BREW_FILE")" != "0" ]]
-then
+check-run-command-as-root() {
   case "$HOMEBREW_COMMAND" in
-    analytics|install|reinstall|postinstall|link|pin|update|upgrade|vendor-install|create|migrate|tap|tap-pin|switch)
+    analytics|install|reinstall|postinstall|link|pin|update|upgrade|\
+    vendor-install|create|migrate|tap|tap-pin|switch)
+      ;;
+    *)
+      return
+      ;;
+  esac
+
+  [[ "$(id -u)" = 0 ]] || return
+
+  local brew_file_ls_info=($(ls -nd "$HOMEBREW_BREW_FILE"))
```

Comment:
I keep to use `ls`. The reasons are:
* It would mean shellout only once instead of twice.
* It avoids surprised behaviors like a failed `stat` outputs something we thought it's success.